### PR TITLE
Security

### PR DESCRIPTION
**Pull request by nopbyte**
This PR includes the functionality to update the password for the user that is currently logged in, and for other users in the sdk. It also lints the code in the examples and removes unused variables.

### CODE REVIEW

File: examples/idm/client-authentication.js [hunk 1]
```diff
@@ -1,31 +1,29 @@
 var token = require('./token_conf');
-var api = 'http://resin.local:8080'
+var api = 'http://resin.local:8080';
 var idmurl = 'http://resin.local:3000';
-//var idmurl = 'http://resin.local:8080'
+// var idmurl = 'http://resin.local:8080'
```
Comment (craigmulligan) [inline examples/idm/client-authentication.js]: nit: remove?

File: src/idm/user.js [hunk 2]
```diff
@@ -109,8 +109,56 @@ const idm = (base, token) => {
...
+    * @public
+    * @function
+    * @memberof agile.idm.user
+    * @param {String} userName user name
+    * @param {String} authType authentication type
+    * @param {String} newPassword new password
+    * @fulfil {Undefined}
+    * @returns {Promise}
+    * @example
+    * agile.idm.user.setPassword('bob','agile-local',"myNewPassword").then(function() {
```
Comment (craigmulligan) [inline src/idm/user.js:131]: Nit: Example mixes single and double quotes

File: src/idm/user.js [hunk 3]
```diff
@@ -109,8 +109,56 @@ const idm = (base, token) => {
...
+    *   console.log('password updated!');
+    * });
+    **/
+    resetPassword: (userName, authType, newPassword) => {
+      return instance.request({
+        method: 'PUT',
+        url: `${base}/api/v1/user/${userName}!@!${authType}/password`,
+        data: {'new_password': newPassword}
+      })
+      .then(res => (res))
```
Comment (craigmulligan) [inline src/idm/user.js]: Wouldn't we prefer to return `res.data`? Or are you using the other bits of the response?

File: src/idm/user.js [hunk 4]
```diff
@@ -109,8 +109,56 @@ const idm = (base, token) => {
...
+    * @example
+    * agile.idm.user.updatePassword("myOldPassword","myNewPassword").then(function() {
+    *   console.log('password updated!');
+    * });
+    **/
+    updatePassword: (oldPassword, newPassword) => {
+      return instance.request({
+        method: 'PUT',
+        url: `${base}/api/v1/user/password`,
+        data: {'old_password': oldPassword, 'new_password': newPassword}
```
Comment (craigmulligan) [inline src/idm/user.js]: Nit: Object keys don't need strings here, axios serializes JavaScript objects to JSON for put/post.

File: src/idm/user.js [hunk 5]
```diff
@@ -109,8 +109,56 @@ const idm = (base, token) => {
...
+    * @name updatePassword
+    * @public
+    * @function
+    * @memberof agile.idm.user
+    * @param {String} oldPassword old password
+    * @param {String} newPassword new password
+    * @fulfil {Undefined}
+    * @returns {Promise}
+    * @example
+    * agile.idm.user.updatePassword("myOldPassword","myNewPassword").then(function() {
```
Comment (craigmulligan) [inline src/idm/user.js:157]: Nit: mixing double/single quotes.

Comment (nopbyte) [reply src/idm/user.js:157]: Ok. Updated it everywhere inside idm to be consistent too.

File: src/idm/user.js (hunk 2 above)
Comment (nopbyte) [reply src/idm/user.js:131]: Ok. also updated all other examples to use single quotes only

**Comment by nopbyte** [comment]
@craig-mulligan, can you have another look please? thanks for the prompt review :)

File: src/idm/group.js [hunk 6]
```diff
@@ -51,7 +51,7 @@ const group = (base, token) => {
     create: (name) => instance.request({
       method: 'POST',
       url: `${base}/api/v1/group/`,
-      data: {'group_name': name}
+      data: {group_name: name}
```
Comment (craig-mulligan) [inline src/idm/group.js]: Nit: spacing :)

Comment (craigmulligan) [inline src/idm/group.js]: Nit: spacing :)

**Review by craigmulligan — APPROVED** [review]

**Comment by craigmulligan** [comment]
Oh also could you make sure to squash commits into a single commit before merging?

And make the message conform to template:
```
My new feature

Connects to #<issue-number>

Change-Type: minor
```

**Comment by nopbyte** [comment]
I did the changes for the spacing across my files.
Could you please  squash it and merge it please?  I can squash the commits but I would prefer to merge directly (not rebase and then merge the PR) for simplicity.

The message I would recommend would be:

Adds the calls to reset and update user's password. Clean the details on the security calls to axios to comply with coding standards for the rest of the project and clean up examples.

There is no issue related to this update

**Comment by craigmulligan** [comment]
merged, tagged, published @nopbyte

**Pull request merged**